### PR TITLE
UCP/STREAM/CUDA: enable cuda buffers in stream api

### PR DESCRIPTION
**Pull request by bureddy**

### CODE REVIEW

**Comment by bureddy** [comment]
@Akshay-Venkatesh @evgeny-leksikov please review

File: src/ucp/stream/stream_recv.c [hunk 1]
```diff
@@ -213,8 +218,9 @@ ucp_stream_process_rdesc(ucp_recv_desc_t *rdesc, ucp_ep_ext_proto_t *ep_ext,
 }
 
 static UCS_F_ALWAYS_INLINE void
-ucp_stream_recv_request_init(ucp_request_t *req, void *buffer, size_t count,
-                             size_t length, ucp_datatype_t datatype,
+ucp_stream_recv_request_init(ucp_ep_h ep, ucp_request_t *req, void *buffer,
```
Comment (Akshay-Venkatesh) [inline src/ucp/stream/stream_recv.c]: Should the function signature similar to other request init functions? Like these two:

https://github.com/openucx/ucx/blob/master/src/ucp/tag/tag_send.c#L119
https://github.com/openucx/ucx/blob/master/src/ucp/stream/stream_send.c#L27

Request is the first parameter, then ep in both the above cases.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7358/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
failure in cuda test:
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7358/label=vulcan-jenkins,worker=0/console
```
20:55:30 +ucp_contig_cuda_stream_lat/45000--+---------+----------+----------+-----------+-----------+
20:55:30 [vulcan-jenkins:34016:0:34016] Caught signal 7 (Bus error: nonexistent physical address)
20:55:30 ==== backtrace ====
20:55:30  0 0x000000000000194b memcpy_uncached_store_avx()  ???:0
20:55:30  1 0x00000000000010a9 gdr_copy_to_bar()  ???:0
20:55:30  2 0x00000000000612f9 uct_gdr_copy_ep_put_short()  /scrap/jenkins/workspace/hpc-ucx-pr/label/vulcan-jenkins/worker/0/contrib/../src/uct/cuda/gdr_copy/gdr_copy_ep.c:45
20:55:30  3 0x0000000000018518 uct_ep_put_short()  /scrap/jenkins/workspace/hpc-ucx-pr/label/vulcan-jenkins/worker/0/contrib/../src/uct/api/uct.h:1723
20:55:30  4 0x000000000003531f ucp_request_recv_data_unpack()  /scrap/jenkins/workspace/hpc-ucx-pr/label/vulcan-jenkins/worker/0/contrib/../src/ucp/core/ucp_request.inl:435
20:55:30  5 0x0000000000450514 ucp_perf_test_runner<(ucx_perf_cmd_t)9, (ucx_perf_test_type_t)0, 0u>::recv_stream()  /scrap/jenkins/workspace/hpc-ucx-pr/label/vulcan-jenkins/worker/0/contrib/../src/tools/perf/ucp_tests.cc:456
20:55:30  6 0x0000000000442c01 ucp_perf_test_runner<(ucx_perf_cmd_t)9, (ucx_perf_test_type_t)0, 0u>::run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/vulcan-jenkins/worker/0/contrib/../src/tools/perf/ucp_tests.cc:416
20:55:30  7 0x0000000000442c01 ucp_perf_test_dispatch()  /scrap/jenkins/workspace/hpc-ucx-pr/label/vulcan-jenkins/worker/0/contrib/../src/tools/perf/ucp_tests.cc:548
20:55:30  8 0x000000000040a5c0 ucx_perf_run()  /scrap/jenkins/workspace/hpc-ucx-pr/label/vulcan-jenkins/worker/0/contrib/../src/tools/perf/libperf.c:1507
20:55:30  9 0x00000000004061dd run_test_recurs()  perftest.c:0
20:55:30 10 0x00000000004062c2 run_test_recurs()  perftest.c:0
20:55:30 11 0x00000000004062c2 run_test_recurs()  perftest.c:0
20:55:30 12 0x00000000004063e7 run_test()  perftest.c:0
20:55:30 13 0x000000000040655c main()  ???:0
20:55:30 14 0x0000000000021c05 __libc_start_main()  ???:0
20:55:30 15 0x00000000004036be _start()  ???:0
20:55:30 ===================
```

**Comment by bureddy** [comment]
@yosefe yes. noticed this. Looks like issue in GDR COPY.  checking with NVIDIA gdr copy team.

**Comment by Akshay-Venkatesh** [comment]
@bureddy As we noticed that the issue goes away with a newer gdrcopy version. Did the bug go away on the test machines?
@yosefe

**Comment by bureddy** [comment]
@Akshay-Venkatesh we haven't updated gdrcopy modules on jenkins yet.

**Comment by bureddy** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7661/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7671/ for details (Mellanox internal link).

**Comment by bureddy** [comment]
@yosefe @evgeny-leksikov please review.
jenkins is fixed with updated gdrcopy.

**Pull request merged**